interfaces/opensearch_client/v0/README.md

```diff
@@ -0,0 +1,82 @@
+# `opensearch_client`
+
+## Usage
+
+This relation interface describes the expected behaviour of any charm interfacing with the [Charmed Opensearch Operator](https://github.com/canonical/opensearch-operator) using the `opensearch-client` relation.
+
+In most cases, this will be accomplished using the [data_interfaces library](https://github.com/canonical/data-platform-libs/blob/main/lib/charms/data_platform_libs/v0/data_interfaces.py), although charm developers are free to provide alternative libraries as long as they fulfil the behavioural and schematic requirements described in this document.
+
+## Direction
+
+```mermaid
+flowchart TD
+    Requirer -- database, \nextra-user-roles --> Provider
+    Provider -- username, \npassword, \nendpoints --> Requirer
+```
+
+As with all Juju relations, the `opensearch-client` interface consists of two parties: a Provider (opensearch charm), and a Requirer (application charm). The Requirer will be expected to provide an index name, and the Provider will provide new unique credentials (along with other optional fields), which can be used to access the index itself.
+
+## Behavior
+
+Both the Requirer and the Provider need to adhere to criteria to be considered compatible with the interface.
+
+### Provider
+- Is expected to create an application user inside the opensearch cluster when the requirer provides the `index` field.
+  - This user is removed when the relation is removed.
+- Is expected to provide `username` and `password` fields when Requirer provides the `index` field.
```

Comment:
Should we specify leveraging juju secrets for this?

Reply:
Definitely - this charm is essentially "green field" so we should be using secrets from the get-go.

Reply:
Updated to include juju secrets for these values, and the TLS CA.